### PR TITLE
Build HDF5 with c++ bindings and the high level interface

### PR DESCRIPTION
**Pull request by tmadlener**
Necessary for building the LUXE simulation that currently produces HDF5 output.

Not sure if we can already use `require` for the release.

### CODE REVIEW

File: environments/key4hep-common/packages.yaml [hunk 1]
```diff
@@ -60,6 +60,9 @@ packages:
     variants: ~openmp
   gperftools:
     variants: ~libunwind
+  hdf5:
+    variants: +cxx+hl
+    require: "+cxx+hl"
```
Comment (jmcarcell) [inline environments/key4hep-common/packages.yaml:65]: Isn't the `require` redundant? It doesn't need to be specified when variants is specified.

Comment (vvolkl) [reply environments/key4hep-common/packages.yaml:65]: No, the concretiser may override this unless it's under spec or require like this

Comment (tmadlener) [reply environments/key4hep-common/packages.yaml:65]: Maybe we could switch all the `variants` to `require`? I would have to check when that became available.

File: environments/key4hep-common/packages.yaml [hunk 2]
```diff
@@ -60,6 +60,9 @@ packages:
     variants: ~openmp
   gperftools:
     variants: ~libunwind
+  hdf5:
+    variants: +cxx+hl
```
Comment (jmcarcell) [inline environments/key4hep-common/packages.yaml:64]:
```suggestion
```

File: environments/key4hep-common/packages.yaml (hunk 1 above)
Comment (jmcarcell) [inline environments/key4hep-common/packages.yaml:65]:
```suggestion
    require: +cxx+hl
```

Comment (jmcarcell) [reply environments/key4hep-common/packages.yaml:65]: > Maybe we could switch all the `variants` to `require`? I would have to check when that became available.

Sounds good

**Pull request merged**